admin/views/Openedx_Woocommerce_Plugin_Settings.php

```diff
@@ -0,0 +1,266 @@
+            platform.</p>
+
+    <?php
+    }
+
+    /**
+     * Output the client secret settings field.
+     *
+     * Retrieves the saved client secret value and outputs a password input field and description text.
+     *
+     * @return void
+     */
+    public function openedx_client_secret_callback()
+    {
+
+        $value = get_option( 'openedx-client-secret' ); ?>
+
+        <input type="text" name="openedx-client-secret" id="openedx-client-secret" 
+            value="<?php echo esc_attr( $value ); ?>" required />
+
+        <p class="description">
+            Confidential key for OAuth application in your Open edX platform.
+        </p>
+
+    <?php
+    }
+
+    /** 
+     * Output the JWT token settings field.
+     *
+     * Retrieves the saved JWT token value and outputs a text input field, generate token button,
+     * and description text.
+     *
+     * @return void
+     */
+    public function openedx_jwt_token_callback()
+    {
+
+        $value = "12121adsad112ad";
+        $masked_value = str_repeat('*', strlen($value) - 4) . substr($value, -4); ?>
+
+        <div class="openedx-jwt-token-wrapper">
+
+            <input class="openedx-jwt-token-input" type="text" name="openedx-jwt-token" id="openedx-jwt-token"
+                value="<?php echo esc_attr($masked_value); ?>" disabled/>
+
+            <button class="button" type="button" id="generate-jwt-token">Generate JWT Token</button>
+
+        </div>
+
+        <p class="description"> Select the Generate Token button to obtain a JWT Token. </p>
+
+    <?php
+    }
+
+    /**
+     * Output introductory text for the settings section.
+     *
+     * Echoes text prompting the user to fill in and save the settings.
+     * 
+     * @return void
+     */
+    public function openedx_settings_section_callback()
+    {
+        printf( 'Configuring the necessary parameters here to establish
+        the connection between this plugin and your Open edX platform.');
+    }
+
+    /**
+     * Sanitize input string to only contain alphanumeric characters.
+     *
+     * This removes any non-alphanumeric characters from the given input string.
+     *
+     * @param string $input The input string to sanitize.
+     * @return string The sanitized input string containing only alphanumeric characters.
+     */
+    public function custom_sanitize_alphanumeric($input) 
+    {
+        return preg_replace('/[^a-zA-Z0-9]/', '', sanitize_text_field($input));
+    }
+
+    /**
+     * Sanitize and validate the Open edX instance URL input.
+     *
+     * This sanitizes the input URL, checks if it is a valid http/https URL, 
+     * and returns the sanitized URL string on success or an empty string on failure.
+     *
+     * @param string $input The URL input to sanitize.
+     * @return string The sanitized URL string on success, empty string on failure.
+     */
+    public function custom_sanitize_url($input) 
```

Comment:
Do you think we need this function? Is there a package or something that already sanitizes URLs?

cc @andres-arango-aptitude

Reply:
I think we can try using the sanitize_url() function, I will test it and then make a comment here to let you know if it works or if I have to look for another function or package.

Reply:
To update about using the sanitize_url() function I just tried it and it works effectively, I will still wait for Andres's opinion about my approach or if there is a better alternative.

Reply:
For me, it's okay you use the sanitize_url()

Reply:
Done.

Reply:
Yes [sanitize_url](https://developer.wordpress.org/reference/functions/sanitize_url/ ) from the documentation has enough validation for your case and will work fine